dot/network/README.md

```diff
@@ -0,0 +1,116 @@
+# Gossamer `network` Package
+
+This package emulates the [peer-to-peer networking capabilities](https://crates.parity.io/sc_network/index.html)
+provided by the [Substrate](https://docs.substrate.io/) framework for blockchain development, which implies that it is
+built on the extensible [`libp2p` networking stack](https://docs.libp2p.io/introduction/what-is-libp2p/). `libp2p`
+provides implementations of a number of battle-tested peer-to-peer (P2P) networking protocols (e.g. [Noise](#noise) for
+[key exchange](#identities--key-management), and [Yamux](#yamux) for [multiplexing](#multiplexing)), and also makes it
+possible to implement the blockchain-specific protocols defined by Substrate (e.g. authoring and finalising blocks, and
+maintaining the [transaction pool](https://docs.substrate.io/v3/concepts/tx-pool/)). The purpose of this document is to
+provide the information that is needed to understand the P2P networking capabilities that are implemented by Gossamer -
+this includes an introduction to P2P networks and `libp2p`, as well as detailed descriptions of the Gossamer P2P
+networking protocols.
+
+## Peer-to-Peer Networking & `libp2p`
+
+[Peer-to-peer](https://en.wikipedia.org/wiki/Peer-to-peer) networking has been a dynamic field of research for over two
+decades, and P2P protocols are at the heart of many blockchain networks. P2P networks can be contrasted with traditional
+[client-server](https://en.wikipedia.org/wiki/Client%E2%80%93server_model) networks where there is a clear separation of
+authority and privilege between the maintainers of the network and its users - in a P2P network, each participant
+possesses equal authority and equal privilege. `libp2p` is a framework for implementing P2P networks that was
+modularized out of [IPFS](https://ipfs.io/); there are implementations in many languages including Go (used by this
+project), Rust, Javascript, C++, and more. In addition to the standard library of protocols in a `libp2p`
+implementation, there is a rich ecosystem of P2P networking packages that work with the pluggable architecture of
+`libp2p`. In some cases, Gossamer uses the `libp2p` networking primitives to implement custom protocols for
+blockchain-specific use cases. What follows is an exploration into three concepts that underpin P2P networks: identity &
+key management, peer discovery & management, and multiplexing.
+
+### Identity & Key Management
+
+Many peer-to-peer networks, including those built with Gossamer, use
+[public-key cryptography](https://en.wikipedia.org/wiki/Public-key_cryptography) (also known as asymmetric cryptography)
+to allow network participants to securely identify themselves and interact with one another. The term "asymmetric"
+refers to the fact that in a public-key cryptography system, each participant's identity is associated with a set of two
```

Comment:
Maybe contrast asymmetric encryption where 2 keys are needed to symmetric encryption where there is only one key to highlight the difference between the two kinds of encryption

Reply:
I think I would rather keep the discussion tightly focused on asymmetric encryption as opposed to introducing new concepts that don't actually pertain to Gossamer.